WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -69,6 +62,15 @@ def index
       @competitions = @competitions.select { |competition| competition.contains?(params[:search]) }
     end
 
+    unless params[:event_ids].empty?
+      @competitions = @competitions.select do |competition|
+        (competition.eventSpecs.split & params[:event_ids]).any?
+      end.sort_by do |competition|
+        -(competition.eventSpecs.split & params[:event_ids]).count
```

Comment:
> I still don't like that the fact that knowledge that "eventSpecs is a space separated string" is leaking out of the Competition model, though. How about creating a method in Competition.rb called matching_event_count(event_ids)?

Yeah, `eventSpecs` should be `private` to begin with. :+1: for moving code which operates on it to `Competition`. We/I don't want to hunt down usages of it if we change it to a proper array.